mobile/lib/src/presentation/pages/sleep_history_page/sleep_history_page.dart

```diff
@@ -1,56 +1,50 @@
+import 'package:polydodo/src/presentation/pages/sleep_history_page/sleep_history_list.dart';
 
 class SleepHistoryPage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
+    final historyCubit = BlocProvider.of<SleepSequenceHistoryCubit>(context);
+
     return Scaffold(
-      appBar: AppBar(
-        backgroundColor: Colors.transparent,
-        shadowColor: Colors.transparent,
-        centerTitle: true,
-        iconTheme: IconThemeData(color: Colors.black),
-        title: Text(
-          'History',
-          style: TextStyle(
-            fontSize: 20,
-            fontWeight: FontWeight.bold,
-            color: Colors.indigo,
-          ),
-        ),
-      ),
-      drawer: NavDrawer(activeTab: NavdrawerTab.History),
-      body: BlocConsumer<SleepHistoryCubit, SleepHistoryState>(
-        listener: (context, state) {
+        appBar: buildAppBar(historyCubit),
+        drawer: NavDrawer(activeTab: NavdrawerTab.History),
+        body:
+            BlocConsumer<SleepSequenceHistoryCubit, SleepSequenceHistoryState>(
+                listener: (context, state) {
           print(state.runtimeType);
-        },
-        builder: (context, state) {
-          if (state is SleepHistoryLoaded) {
-            return ListView.builder(
-                itemCount: state.history.length,
-                itemBuilder: (context, index) {
-                  return Card(
-                      child: ListTile(
-                    onTap: () => {
-                      BlocProvider.of<SleepHistoryCubit>(context)
-                          .selectNight(state.history[index]),
-                      ExtendedNavigator.of(context)
-                          .replace(Routes.nightStatsPage)
-                    },
-                    title: Text(state.history[index].id.toString()),
-                  ));
-                });
+        }, builder: (context, state) {
+          if (state is SleepSequenceHistoryLoaded ||
+              state is SleepSequenceHistoryEditInProgress) {
+            return buildHistoryList(context, state, historyCubit);
           } else {
             return Container();
           }
-        },
-      ),
-    );
+        }),
+        floatingActionButton: _buildFloatingActionButton(historyCubit));
   }
 }
+
+Widget _buildFloatingActionButton(var historyCubit) {
+  return BlocConsumer<SleepSequenceHistoryCubit, SleepSequenceHistoryState>(
+      listener: (context, state) => {},
+      builder: (context, state) {
+        if (state is SleepSequenceHistoryEditInProgress) {
+          return Visibility(
+              visible: (state.selectedSequences?.isNotEmpty ?? false),
+              child: FloatingActionButton(
+                onPressed: () => historyCubit.deleteSelected(),
+                child: Icon(Icons.delete),
+                backgroundColor: Colors.red,
+              ));
+        } else {
+          return Container();
+        }
```

Comment:
ternary operator (basically for simple value returns, use ternaries and for more complex stuff use if else)